internal/wire/wire.go

```diff
@@ -119,123 +119,142 @@ func Execute(context interface{}, mgoDB *db.DB, graphDB *cayley.Handle, viewPara
 // validateStartType verifies the start type of a view path.
 func validateStartType(context interface{}, db *db.DB, v *view.View) error {
 
-	// Declare variables to track the first level relationship predicate.
-	var firstRel string
-	var firstDir string
-
-	// Extract the first level relationship predicate.
-	for _, segment := range v.Path {
-		if segment.Level == 1 {
-			firstRel = segment.Predicate
-			firstDir = segment.Direction
+	// Loop over paths to validate the start type for each.
+PathLoop:
```

Comment:
I think this is a good solution for this case, because it saves additional work in that inner loop compared to having a boolean variable or something.